### DISCUSSION TITLE
Normalize white/black balance in image sequence to reduce flickering.

### DISCUSSION THREAD

**Discussion by trig-vee** [ImageMagick/ImageMagick#4523] [Digital Image Processing] [answered]
My project generates a GIF from an image sequence extracted from "real world" photographs. Because of imperfections in real world lighting, the source images have slightly different background colors. This leads to some unwanted "flickering" in the resulting GIF:

![Animation](https://user-images.githubusercontent.com/6199970/143978224-84c7c802-8ba7-48cc-9dc7-254f2887b817.gif)

[Zip - source image sequence](https://github.com/ImageMagick/ImageMagick/files/7622609/sequence-frames.zip)

Is there a way I could analyze _across_ an image sequence, then use the information to adjust each individual frame in a way that  reduces flickering in the final GIF?

Thanks, I've been struggling to solve this and would appreciate any ideas or references.

**Comment by fmw42** [comment, marked as answer]
What version of ImageMagick? What platform? If I assume a Unix-like system running IM 6, then this works to do division normalization using sub-shell processing to avoid writing temp files.

```
cd
cd desktop/sequence-frames
(
for img in *.png; do
convert "$img" \( +clone -blur 0x30 \) +swap -compose divide -composite miff:-
done
) | convert -delay 10 - -layers optimize anim.gif

```

![anim](https://user-images.githubusercontent.com/13458218/143990393-2ce35581-b1ac-4a42-9b41-de3c8de44d6f.gif)

**Reply by trig-vee** [reply]
@fmw42 yes, I am developing on a Unix machine. 

Thank you! I did a quick test and this looks like what I need. I will look more into [Divide.](https://legacy.imagemagick.org/Usage/compose/#divide)
